src/core/tiledmesh/qgstiledmeshboundingvolume.cpp

```diff
@@ -41,23 +42,37 @@ Qgis::TiledMeshBoundingVolumeType QgsTiledMeshNodeBoundingVolumeRegion::type() c
 
 QgsBox3D QgsTiledMeshNodeBoundingVolumeRegion::bounds( const QgsCoordinateTransform &transform, Qgis::TransformDirection direction ) const
 {
-  if ( !transform.isValid() || transform.isShortCircuited() )
-  {
-    return mRegion;
-  }
-  else
+  if ( !mTransform.isIdentity() || ( transform.isValid() && !transform.isShortCircuited() ) )
   {
     // transform each corner of the box, then collect the min/max x/y/z values of the result
-    QVector< double > x{ mRegion.xMinimum(), mRegion.xMinimum(), mRegion.xMaximum(), mRegion.xMaximum(), mRegion.xMinimum(), mRegion.xMinimum(), mRegion.xMaximum(), mRegion.xMaximum() };
-    QVector< double > y{ mRegion.yMinimum(), mRegion.yMaximum(), mRegion.yMinimum(), mRegion.yMaximum(), mRegion.yMinimum(), mRegion.yMaximum(), mRegion.yMinimum(), mRegion.yMaximum() };
-    QVector< double > z{ mRegion.zMinimum(), mRegion.zMinimum(), mRegion.zMinimum(), mRegion.zMinimum(), mRegion.zMaximum(), mRegion.zMaximum(), mRegion.zMaximum(), mRegion.zMaximum() };
+    QVector<QgsVector3D > corners = mRegion.corners();
+    QVector< double > x;
+    x.reserve( 8 );
+    QVector< double > y;
+    y.reserve( 8 );
+    QVector< double > z;
+    z.reserve( 8 );
+    for ( int i = 0; i < 8; ++i )
+    {
+      QgsVector3D corner = corners[i];
+      if ( !mTransform.isIdentity() )
+        corner = mTransform.map( corner );
```

Comment:
I think regions do not get transform applied - see https://github.com/CesiumGS/3d-tiles/tree/main/specification#tile-transforms:


The transform property applies to [...] tile.boundingVolume, except when tile.boundingVolume.region is defined, which is explicitly in EPSG:4979 coordinates.

Reply:
Thanks, fixed (also fixed the sphere transform radius handling)